asgiref/sync.py

```diff
@@ -337,5 +339,33 @@ def get_current_task():
 
 
 # Lowercase is more sensible for most things
-sync_to_async = SyncToAsync
 async_to_sync = AsyncToSync
+
+
+def sync_to_async(sync_thing, impl=SyncToAsync, thread_sensitive=False):
+    """
+    Convert a sync "thing" into an async version of it.
+
+    :param sync_thing:
+        Any one of the following -
+
+            1. Function
+            2. Iterable (any object that implements ``__iter__`` or ``__getitem__``)
+            3. Generator function
+
+    :param impl:
+        The ``SyncToAsync`` implementation to use.
+
+    :param thread_sensitive:
+        Passed to ``cls``.
+    """
+
+    is_iterable = hasattr(sync_thing, "__iter__") or hasattr(sync_thing, "__getitem__")
+    if is_iterable:
+        return sync_iterable_to_async(sync_thing, impl, thread_sensitive)
+
+    is_generator = inspect.isgeneratorfunction(sync_thing)
```

Comment:
Inspect is traditionally a slow module. Have you run speed tests to see what the impact here is on the performance of sync_to_async? It's already not the best function due to the threading, but I want to avoid making it substantially slower,